### PR TITLE
[improvement](agg) Do not serialize bitmap to string

### PR DESCRIPTION
**Pull request by mrhhsg**
## Proposed changes

There is no need to serialize bitmap into a string because bitmap can be serialized into protobuf.

T his PR need to pick to 2.0.1 for user case.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by mrhhsg** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 48.11 seconds
 stream load tsv:          543 seconds loaded 74807831229 Bytes, about 131 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.1 seconds inserted 10000000 Rows, about 343K ops/s
 storage size: 17162263780 Bytes

**Comment by mrhhsg** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.2 seconds
 stream load tsv:          545 seconds loaded 74807831229 Bytes, about 130 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17162263326 Bytes

File: fe/fe-common/src/main/java/org/apache/doris/common/Config.java [hunk 1]
```diff
@@ -1619,7 +1619,7 @@ public class Config extends ConfigBase {
      * Max data version of backends serialize block.
      */
     @ConfField(mutable = false)
-    public static int max_be_exec_version = 2;
+    public static int max_be_exec_version = 3;
```
Comment (yiguolei) [inline fe/fe-common/src/main/java/org/apache/doris/common/Config.java:1622]: Add comment here, why need change be exec version to 3??

**Review by yiguolei — DISMISSED** [review]
LGTM

File: be/src/vec/aggregate_functions/aggregate_function_bitmap.h [hunk 2]
```diff
@@ -146,10 +146,127 @@ struct AggregateFunctionBitmapData {
...
+            : IAggregateFunctionDataHelper<Data, Derived>(argument_types_) {}
+
+    void serialize_to_column(const std::vector<AggregateDataPtr>& places, size_t offset,
+                             MutableColumnPtr& dst, const size_t num_rows) const override {
+        if (version >= 3) {
+            auto& col = assert_cast<ColumnBitmap&>(*dst);
+            col.resize(num_rows);
+            auto* data = col.get_data().data();
+            for (size_t i = 0; i != num_rows; ++i) {
+                data[i] = this->data(places[i] + offset).value;
```
Comment (HappenLee) [inline be/src/vec/aggregate_functions/aggregate_function_bitmap.h:165]: better use `std::move(this->data(places[i] + offset).value)`

File: be/src/vec/aggregate_functions/aggregate_function_bitmap.h [hunk 3]
```diff
@@ -146,10 +146,127 @@ struct AggregateFunctionBitmapData {
...
+        }
+    }
+
+    void serialize_without_key_to_column(ConstAggregateDataPtr __restrict place,
+                                         IColumn& to) const override {
+        if (version >= 3) {
+            auto& col = assert_cast<ColumnBitmap&>(to);
+            size_t old_size = col.size();
+            col.resize(old_size + 1);
+            col.get_data()[old_size] = this->data(place).value;
```
Comment (HappenLee) [inline be/src/vec/aggregate_functions/aggregate_function_bitmap.h:240]: seems better use std::move?

**Comment by mrhhsg** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.42 seconds
 stream load tsv:          542 seconds loaded 74807831229 Bytes, about 131 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17162303525 Bytes

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**